forem/forem
# add lightbox fullscreen functionality for code

## What type of PR is this? (check all applicable)


## Description

I added fullscreen functionality for code blocks via panel, shown on hover. There are some updates for highlight code block, which allow division of concerns in code:

* classnames starting with _js-_ for event-listeners - in order to be independent from markup.
* added 'highlight__panel-action', which allow to add several elements (action-icons, e.g. copy) by the same pattern, which will result in proper stylinh
* added store of window scroll position and preventing scroll on document on fullscreen mode
* added event-listener for escape key on keyboard for switching off the full-screen
* scss is written in BEM, using some utility classes.
* I can't help to auto-fix ruby tests for travis-cli - in last commit. It is definitely not related to initial scope, but can help (or can it be removed from this PR).

## Related Tickets & Documents

Closes #8990 

## QA Instructions, Screenshots, Recordings

* add different code blocks to the page
* hover over code blocks to see the fullscreen button
* click on button to enter fullscreen mode
* click on button back / press 'escape' button

## Added tests?


## Added to documentation?



PS: I hope this pull-requst will be helpful:)
